lib/dialects/abstract/query-generator.js

```diff
@@ -1455,8 +1448,18 @@ var QueryGenerator = {
       if (options.where) {
         if (subQuery) {
           subQueryItems.push(' WHERE ' + options.where);
+
         } else {
           mainQueryItems.push(' WHERE ' + options.where);
+          // Walk the array backwards to find the last instance
+          for(var i = mainQueryItems.length - 1; i >= 0; --i) {
+            if(mainQueryItems[i].match(/^SELECT/)) {
+              var string = this.selectSqlFragmentBuilder(options, model, mainAttributes, table, mainTableAs, options.where);
+              //console.log(`Replacing '${mainQueryItems[0]}' with '${string}'`)
+              mainQueryItems[i] = string;
+              break;
+            }
```

Comment:
You are correct, more or less about what this is for. It was originally designed to handle nesting queries, but since that's handled already by the recursive subquery calls, I'll simplify it.